### PR TITLE
Test date extr

### PR DESCRIPTION
**Pull request by herector8**

### CODE REVIEW

File: test/test_dates.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    ('*2011/12/01 08:15:27.123456_blabla.xlsx', dt.datetime(year=2011, month=12, day=1, hour=8, minute=15, second=27, microsecond=123456)),
+    ('*2011-12-01 08:15:27.123456_blabla.xlsx', dt.datetime(year=2011, month=12, day=1, hour=8, minute=15, second=27, microsecond=123456)),
+    ('*2011.12.01 08:15:27.123456_blabla.xlsx', dt.datetime(year=2011, month=12, day=1, hour=8, minute=15, second=27, microsecond=123456)),
+    ('*2011 12 01 08:15:27.123456_blabla.xlsx', dt.datetime(year=2011, month=12, day=1, hour=8, minute=15, second=27, microsecond=123456)),
+    ('*20111201 08:15:27.123456_blabla.xlsx', dt.datetime(year=2011, month=12, day=1, hour=8, minute=15, second=27, microsecond=123456)),
+    ('*10-12-2011 08:15:27.123456_blabla.xlsx', dt.datetime(year=2011, month=10, day=12, hour=8, minute=15, second=27, microsecond=123456)),
+    ('*10-12-2011_bla.txt', dt.datetime(year=2011, month=10, day=12)),
+    (('*2011-12-31_bla.txt', dt.datetime(year=2011, month=12, day=31))),
+    (('*20111231_bla.txt', dt.datetime(year=2011, month=12, day=31)))
+    ])
```
Comment (chrispcharlton) [inline test/test_dates.py:16]: Probably best to move these to either a global variable (something like `test_cases`) just to keep things clean + maintainable i.e. easy to add more test cases. Could be good in future to add test cases with text in front of the date.

File: test/test_dates.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+def test_date_from_string(test_input, expected):
+    assert extract_date_from_str(patterns_list=patterns_list, string=test_input) == expected
```
Comment (chrispcharlton) [inline test/test_dates.py:19]: very clean parametrized test!

File: src/date_extraction.py [hunk 3]
```diff
@@ -29,9 +31,4 @@ def extract_date_from_str(patterns_list, string):
                 return parse(match)
             except:
                 print(f"String {match} was not recognised as date. Check date patterns")
-                return False
-
-
-if __name__ == '__main__':
-    string = 'letter_of_resignation_2011/01/01 08:15:27.123456.xlsx'
-    date = extract_date_from_str(patterns_list=patterns_list, string=string)
+                return False
```
Comment (chrispcharlton) [inline src/date_extraction.py:34]: Probably makes more sense to return `None` than `False`?

**Review by chrispcharlton — CHANGES_REQUESTED** [review]

File: test/test_dates.py (hunk 1 above)
Comment (herector8) [reply test/test_dates.py]: That's a good idea. Have done that

File: src/date_extraction.py (hunk 3 above)
Comment (herector8) [reply src/date_extraction.py]: Agree, have changed that as well

**Review by chrispcharlton — APPROVED** [review]

**Pull request merged**